### PR TITLE
created slider component

### PR DESCRIPTION
**Pull request by AhmedAdel999**
- Created slider component with its various variants(circular, vertical and horizontal).
- Added snapshot testing.

### CODE REVIEW

File: modules/shared/components/atoms/Slider/ISlider.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+declare namespace ISlider {
+  export interface IProps {
+    type: 'circular' | 'horizontal' | 'vertical';
```
Comment (AhmedElshaf3i) [inline modules/shared/components/atoms/Slider/ISlider.d.ts:3]: @Safei-Ashraf  recommend using enum
e.x:
enum TYPE{
  CITCULAR= "circular"
HORIZ= "horizontal"
VERTICAL= "vertical"
}
export interface IProps {
    type: TYPE,...
    }

File: modules/shared/components/atoms/Slider/index.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  });
+  const outerVerticalClasses = classNames('w-2 rounded-sm', {
+    'bg-success': progress === maximumProgress,
+    'bg-primary':
+      progress < maximumProgress && verticalMeterColor === 'primary',
+    'bg-primary-shd5':
+      progress < maximumProgress && verticalMeterColor === 'primary-shd5',
+    'bg-error': progress < maximumProgress && verticalMeterColor === 'error',
+  });
+  const innerVerticalClasses = classNames('w-2 bg-white rounded-t-sm');
```
[the remark this answers is not in the document]
Comment (AhmedAdel999) [reply modules/shared/components/atoms/Slider/index.tsx:29]: I checked it again and it looks fine to me not sure what you mean here

File: modules/shared/components/atoms/Slider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  progress,
+  type,
+  height,
+  verticalMeterColor,
+  radius,
+}): ReactElement => {
+  const maximumProgress = 100;
+  const halfMaximumProgress = 50;
+  const defaultHeight = 300;
+  const outerHorizontalClasses = classNames('w-full h-2 rounded-sm bg-white');
```
[the remark this answers is not in the document]
Comment (AhmedAdel999) [reply modules/shared/components/atoms/Slider/index.tsx:16]: I guess you are talking about the hidden box around the progress bar, I didn't implement that that's why both heights are equal, but it's true that the height should be 4px.

File: modules/shared/components/atoms/Slider/index.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  const innerHorizontalClasses = classNames('h-2 rounded-sm', {
+    'bg-success': progress === maximumProgress,
+    'bg-primary': progress < maximumProgress,
+  });
+  const outerVerticalClasses = classNames('w-2 rounded-sm', {
```
[the remark this answers is not in the document]
Comment (AhmedAdel999) [reply modules/shared/components/atoms/Slider/index.tsx:21]: Same thing as before

File: modules/shared/components/atoms/Slider/index.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  }
+  if (type === 'vertical') {
+    return (
+      <div
+        style={{ height: `${height ?? defaultHeight}px` }}
+        className={outerVerticalClasses}
+      >
+        <div
+          className={innerVerticalClasses}
+          style={{ height: `${maximumProgress - progress}%` }}
```
[the remark this answers is not in the document]
Comment (AhmedAdel999) [reply modules/shared/components/atoms/Slider/index.tsx:69]: It's already increasing from bottom to top.

**Comment by AhmedAdel999** [comment]
Requested changes done (except for using enum)

**Comment by atwfk** [comment]
@AhmedAdel999  checkout [this file](https://github.com/m3ntorship/pickify-frontend/blob/8a06e206524c2b83e4006542c523b5fa6678d1c1/modules/shared/components/TextInputs/types/ETextInput.ts) and [this](https://github.com/m3ntorship/pickify-frontend/blob/8a06e206524c2b83e4006542c523b5fa6678d1c1/modules/shared/components/TextInputs/types/ITextInputs.d.ts)

**Comment by AhmedAdel999** [comment]
@atwfk What are the benefits of using enum? I feel like it's too much hassle especially that I will only use that enum in just that file

**Comment by atwfk** [comment]
as @elhaw said: `Let us consider if the developer who used the component typed Small instead of small or there is a typo in the word `

**Comment by AhmedAdel999** [comment]
I already solved that by using string literals.

**Comment by atwfk** [comment]
i think both will do the same...what only matters is we don't want the developer to face the capital and small letters issue

**Comment by AhmedElshaf3i** [comment]
LGTM

**Review by AhmedElshaf3i — APPROVED** [review]

**Pull request merged**